src/app/services/cart.service.ts

```diff
@@ -0,0 +1,38 @@
+import { Injectable } from '@angular/core';
+import { Item } from '../models/item';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class CartService {
+
+  constructor() { }
+
+  cart: Item[] = []
+  itemsInCart: Item[] = []
+
+  addItemToCart(item: Item) {
+    if (!this.cart.includes(item)){
+      item.quantity = 1
+      this.cart.push(item) 
+    }
+    else this.cart[this.cart.indexOf(item)].quantity += 1
+  }
+
+  addOneElementToCart(item: Item) {
+    this.cart[this.cart.indexOf(item)].quantity += 1
+  }
+
+  removeOneElementFromCart(item: Item) {
+    if (this.cart[this.cart.indexOf(item)].quantity === 1) this.cart.splice(this.cart.indexOf(item), 1)
```

Comment:
This is more of a personal preference but I think it's much better to always have blocks for conditional logic.